### DISCUSSION TITLE
Is it currently possible to programatically disable the scroll bar when there is nothing to scroll?

### DISCUSSION THREAD

**Discussion by paveloom** [wezterm/wezterm#2590] [Q&A] [answered]
For example, when Neovim is open, the scrollbar is not particularly useful.

**Comment by wez** [comment, marked as answer]
You could do something like:

```lua
wezterm.on('update-status', function(window, pane)
  local overrides = window:get_config_overrides() or {}
  overrides.enable_scroll_bar = not pane:is_alt_screen_active()
  window:set_config_overrides(overrides)     
end)
```

See also:
* https://wezfurlong.org/wezterm/config/lua/pane/is_alt_screen_active.html
* https://wezfurlong.org/wezterm/config/lua/window-events/update-status.html
* https://wezfurlong.org/wezterm/config/lua/window/get_config_overrides.html
* https://wezfurlong.org/wezterm/config/lua/window/set_config_overrides.html

**Reply by paveloom** [reply]
Neat! Thank you.

**Reply by jaminthorns** [reply]
Additionally, if you'd like to disable the scrollbar when there aren't enough rows to actually scroll, you can do that with `pane:get_dimensions()`:

```lua
wezterm.on("update-status", function(window, pane)
  local overrides = window:get_config_overrides() or {}
  local dimensions = pane:get_dimensions()

  overrides.enable_scroll_bar = dimensions.scrollback_rows > dimensions.viewport_rows and not pane:is_alt_screen_active()

  window:set_config_overrides(overrides)
end)
```
